src/main/java/com/solvd/app/dao/DoctorDAO.java

```diff
@@ -0,0 +1,149 @@
+package com.solvd.app.dao;
+
+import com.solvd.app.interfaces.IBaseDAO;
+import com.solvd.app.models.Doctor;
+import com.solvd.app.utils.ConnectionPool;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DoctorDAO implements IBaseDAO<Doctor> {
+    private static final Logger LOGGER = LogManager.getLogger(DoctorDAO.class);
+    private ConnectionPool connectionPool = ConnectionPool.getInstance();
+    private PersonDAO personDAO = new PersonDAO();
+    private SpecialtyDAO specialtyDAO = new SpecialtyDAO();
+
+    public void createEntity(Doctor doctor) {
+        Connection connection = connectionPool.getConnection();
+
+        try {
+            String sql = "INSERT INTO doctors (specialty_id, person_id, years_of_experience) VALUES (?, ?, ?)";
+            PreparedStatement preparedStatement = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS);
+            preparedStatement.setInt(1, doctor.getSpecialty().getSpecialtyID());
+            preparedStatement.setInt(2, doctor.getPerson().getPersonID());
+            preparedStatement.setInt(3, doctor.getYearsOfExperience());
```

Comment:
YearsOfExperience might be better if it's calculated from an actual start date and using a Date type in your db implementation

Reply:
Thank you both for the suggestions. I will keep it like it is for now.